spec/acceptance/nginx_prometheus_exporter_spec.rb

```diff
@@ -0,0 +1,39 @@
+require 'spec_helper_acceptance'
+
+describe 'prometheus nginx exporter' do
+  it 'nginx_prometheus_exporter works idempotently with no errors' do
+    on hosts, puppet('module', 'install', 'puppet-nginx', '--version', '3.2.0')
+    pp = <<-EOS
+  class { 'prometheus::nginx_prometheus_exporter':
+    scrape_uri => 'http://localhost:8888/stub_status',
+  }
+  include nginx
+  nginx::resource::server { 'localhost':
+    listen_port => 8888,
+    locations   => {
+      'stub_status' => {
+        location    => '/stub_status',
+        stub_status => true,
+      }
+    }
+  }
+  Class['nginx'] -> Class['prometheus::nginx_prometheus_exporter']
+  Nginx::Resource::Server['localhost'] -> Class['prometheus::nginx_prometheus_exporter']
+  EOS
+    apply_manifest(pp, catch_failures: true)
+    apply_manifest(pp, catch_changes: true)
+  end
+
+  describe 'default install' do
+    describe service('nginx_prometheus_exporter') do
+      it { is_expected.to be_running }
+      it { is_expected.to be_enabled }
+    end
+    describe port(9113) do
+      it { is_expected.to be_listening.with('tcp6') }
+    end
+    describe process('nginx-prometheus-exporter') do
+      its(:args) { is_expected.to match %r{\ -nginx.scrape-uri http://localhost:8888/stub_status} }
```

Comment:
Maybe this will fix the failing spec?

```suggestion
      its(:args) { is_expected.to match %r{-nginx\.scrape-uri 'http:\/\/localhost:8888\/stub_status'} }
```

Reply:
actually it was a bit weirder.

it seems the `describe process` is executing `/bin/sh -c ps -C nginx-prometheus-exporter -o args= | head -1` and in newer versions of ps the cmd gets truncated to 15 characters.